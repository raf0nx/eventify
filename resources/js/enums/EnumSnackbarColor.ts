export enum EnumSnackbarColor {
    SUCCESS = "deep-purple",
    INFO = "blue",
    WARNING = "orange darken-1",
    ERROR = "red accent-2"
}
