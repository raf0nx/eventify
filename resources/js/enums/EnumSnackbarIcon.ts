export enum EnumSnackbarIcon {
    SUCCESS = "mdi-check-circle",
    INFO = "mdi-information-outline",
    WARNING = "mdi-alert",
    ERROR = "mdi-alert-circle-outline"
}
