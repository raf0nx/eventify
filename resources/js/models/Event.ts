import { User } from "./User";

export interface Event {
    created_at: Date;
    description: string;
    event_image: string;
    id: number;
    name: string;
    start_datetime: Date;
    updated_at: Date;
    users: User[];
}
