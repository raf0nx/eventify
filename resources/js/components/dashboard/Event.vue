<template>
	<v-lazy
		v-model="isActive"
		:options="{
			threshold: 0.5,
		}"
        min-height="200"
		transition="fade-transition"
	>
		<v-card class="my-12" elevation="8">
			<v-img
				lazy-src="https://picsum.photos/id/11/10/6"
				src="https://picsum.photos/id/11/500/300"
				height="300"
			></v-img>

			<v-card-subtitle class="pb-0">
				Event starts on {{ eventStartDate }} at {{ eventStartTime }}
			</v-card-subtitle>

			<v-card-title class="font-weight-bold">{{
				event.name
			}}</v-card-title>

			<v-card-text>
				<div>
					{{ event.description }}
				</div>

				<div class="pt-4">
					{{ event.users.length }} interested &bull; 0 going
				</div>
			</v-card-text>

			<v-card-actions class="py-4">
				<v-btn color="deep-purple" text>
					<v-icon left>mdi-star</v-icon>
					Interested
				</v-btn>
				<v-btn color="deep-purple" class="white--text font-weight-bold">
					<v-icon left>mdi-check-circle</v-icon>
					Going
				</v-btn>

				<v-spacer></v-spacer>

				<v-btn icon>
					<v-tooltip top>
						<template v-slot:activator="{ on, attrs }">
							<v-icon v-on="on" v-bind="attrs">mdi-heart</v-icon>
						</template>
						<span>Add to favourites</span>
					</v-tooltip>
				</v-btn>

				<v-btn icon>
					<v-tooltip top>
						<template v-slot:activator="{ on, attrs }">
							<v-icon v-on="on" v-bind="attrs">
								mdi-share-variant
							</v-icon>
						</template>
						<span>Share</span>
					</v-tooltip>
				</v-btn>
			</v-card-actions>
		</v-card>
	</v-lazy>
</template>

<script lang="ts">
	import { Vue, Component, Prop } from "vue-property-decorator";
	import moment from "moment";

	import { Event as EventModel } from "@/models/Event";

	@Component
	export default class Event extends Vue {
		@Prop()
		event!: EventModel;

        isActive = false;

		get eventStartDate(): string {
			return moment(this.event.start_datetime).format("dddd, MMM Do YYYY");
		}

		get eventStartTime(): string {
			return moment(this.event.start_datetime).format("h a");
		}
	}
</script>

<style>
</style>