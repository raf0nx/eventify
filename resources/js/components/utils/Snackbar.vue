<template>
	<v-snackbar
		v-model="snackbar.showSnackbar"
        class="mb-8"
		app
		fixed
		bottom
		:color="snackbar.color"
	>
		<div class="d-flex align-center">
			<v-icon color="white">{{ snackbar.icon }}</v-icon>
			<span class="font-weight-bold ml-2">{{ snackbar.message }}</span>
		</div>
	</v-snackbar>
</template>

<script lang="ts">
	import { Vue, Component } from "vue-property-decorator";

	import { UtilsModule } from "@modules/Utils";
	import { SnackbarModel } from "@/models/Snackbar";

	@Component
	export default class Snackbar extends Vue {
		get snackbar(): SnackbarModel {
			return UtilsModule.snackbar;
		}
	}
</script>

<style scoped>
</style>