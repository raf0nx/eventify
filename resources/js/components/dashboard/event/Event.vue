<template>
	<v-col class="col-xs-12 col-sm-6 col-lg-4 col-xl-3">
		<v-lazy
			v-model="isActive"
			:options="{
				threshold: 0.5,
			}"
			min-height="240"
			transition="fade-transition"
		>
			<v-hover v-slot="{ hover }">
				<v-card
					ripple
					class="rounded-lg"
					:elevation="hover ? 12 : 4"
					max-height="480"
					@click="goToEventDetails()"
				>
					<v-img
						lazy-src="https://picsum.photos/id/11/10/6"
						src="https://picsum.photos/id/11/500/300"
						max-height="200"
						class="rounded-t-lg"
					></v-img>

					<v-card-subtitle class="pb-0">
						{{ eventStartDate }} at {{ eventStartTime }}
					</v-card-subtitle>

					<v-card-title class="font-weight-bold">
						<router-link
							:to="{ name: 'Event', params: { id: event.id } }"
							class="
								event__link
								black--text
								text-decoration-none text-truncate
							"
							>{{ event.name }}</router-link
						>
					</v-card-title>

					<v-card-text>
						<v-tooltip bottom max-width="320">
							<template v-slot:activator="{ on, attrs }">
								<div
									class="text-truncate"
									v-on="on"
									v-bind="attrs"
								>
									{{ event.description }}
								</div>
							</template>
							<p class="text-justify mb-0">
								{{ event.description }}
							</p>
						</v-tooltip>
						<div class="pt-4">
							{{ event.users.length }} interested &bull; 0 going
						</div>
					</v-card-text>

					<v-card-actions class="pt-2 pb-4">
						<v-btn
							@click.stop="() => true"
							color="deep-purple"
							text
						>
							<v-icon left>mdi-star</v-icon>
							Interested
						</v-btn>
						<v-btn
							@click.stop="() => true"
							color="deep-purple"
							class="white--text font-weight-bold"
						>
							<v-icon left>mdi-check-circle</v-icon>
							Going
						</v-btn>

						<v-spacer></v-spacer>

						<v-btn icon>
							<v-tooltip top>
								<template v-slot:activator="{ on, attrs }">
									<v-icon
										@click.stop="() => true"
										color="red accent-2"
										v-on="on"
										v-bind="attrs"
										>mdi-heart-outline</v-icon
									>
								</template>
								<span>Add to favourites</span>
							</v-tooltip>
						</v-btn>
					</v-card-actions>
				</v-card>
			</v-hover>
		</v-lazy>
	</v-col>
</template>

<script lang="ts">
	import { Vue, Component, Prop } from "vue-property-decorator";
	import moment from "moment";

	import { Event as EventModel } from "@/models/Event";

	@Component
	export default class Event extends Vue {
		@Prop()
		event!: EventModel;

		isActive = false;

		get eventStartDate(): string {
			return moment(this.event.start_datetime).format("dddd, MMM Do YYYY");
		}

		get eventStartTime(): string {
			return moment(this.event.start_datetime).format("h a");
		}

		goToEventDetails(): void {
			setTimeout(() => {
				this.$router.push({
					name: "Event",
					params: { id: String(this.event.id) },
				});
			}, 200);
		}
	}
</script>

<style scoped>
	.event__link {
		border-bottom: 2px solid white;
		transition: border-bottom linear 0.5s;
	}
	.event__link:hover {
		border-bottom: 2px solid black;
	}
</style>