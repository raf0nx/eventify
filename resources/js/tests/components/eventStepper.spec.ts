import { createLocalVue, shallowMount, Wrapper } from "@vue/test-utils";
import Vuetify from "vuetify";
import moment from "moment";
import axios from "axios";

import EventStepper from "@components/dashboard/event/EventStepper.vue";
import { Event } from "@/tests/constans/Event";
import { UtilsModule } from "@/store/modules/Utils";
import { SnackbarModel } from "@/models/Snackbar";
import { EnumSnackbarIcon } from "@/enums/EnumSnackbarIcon";
import { EnumSnackbarColor } from "@/enums/EnumSnackbarColor";

jest.mock("axios");

describe("EventStepper.vue", () => {
    const localVue = createLocalVue();
    let wrapper: Wrapper<EventStepper>;
    let vuetify: Vuetify;

    beforeEach(() => {
        vuetify = new Vuetify();
        wrapper = shallowMount(EventStepper, { localVue, vuetify });
    });

    afterEach(() => {
        jest.clearAllMocks();
        wrapper.destroy();
    });

    it("Should match snapshot", () => {
        // Assert
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("Should get minimum event date", () => {
        // Assert
        // @ts-ignore
        expect(wrapper.vm.minEventDate).toEqual(moment().format("YYYY-MM-DD"));
    });

    it("Should get minimum event time current day", () => {
        // Assert
        // @ts-ignore
        expect(wrapper.vm.minEventTime).toEqual(moment().format("HH:mm"));
    });

    it("Should get minimum event time", () => {
        // Act
        // @ts-ignore
        wrapper.vm.eventDate = "1970-01-01 00:00:00";

        // Assert
        // @ts-ignore
        expect(wrapper.vm.minEventTime).toEqual("00:00");
    });

    it("Should create an event and dispatch notification", async () => {
        // Arrange
        // @ts-ignore
        wrapper.vm.event.name = Event.name;
        // @ts-ignore
        wrapper.vm.event.description = Event.description;
        // @ts-ignore
        wrapper.vm.event.image = Event.image;
        // @ts-ignore
        wrapper.vm.event.start_datetime = Event.start_datetime;

        // Act
        // @ts-ignore
        axios.post.mockImplementationOnce(() => Promise.resolve());
        // @ts-ignore
        await wrapper.vm.createEvent();

        // Assert
        expect(UtilsModule.snackbar).toEqual(
            new SnackbarModel()
                .setShowSnackbar(true)
                .setMessage("Event created successfully!")
                .setIcon(EnumSnackbarIcon.SUCCESS)
                .setColor(EnumSnackbarColor.SUCCESS)
        );
    });

    it("Should fail creating an event and inform user about it", async () => {
        // Arrange
        // @ts-ignore
        wrapper.vm.event.name = Event.name;
        // @ts-ignore
        wrapper.vm.event.description = Event.description;
        // @ts-ignore
        wrapper.vm.event.image = Event.image;
        // @ts-ignore
        wrapper.vm.event.start_datetime = Event.start_datetime;

        // Act
        // @ts-ignore
        axios.post.mockImplementationOnce(() => Promise.reject());
        // @ts-ignore
        await wrapper.vm.createEvent();

        // Assert
        expect(UtilsModule.snackbar).toEqual(
            new SnackbarModel()
                .setShowSnackbar(true)
                .setMessage(
                    "Couldn't create your event. Reload the page and try again."
                )
                .setIcon(EnumSnackbarIcon.ERROR)
                .setColor(EnumSnackbarColor.ERROR)
        );
    });
});
