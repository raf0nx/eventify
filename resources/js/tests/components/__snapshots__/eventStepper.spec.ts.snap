// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventStepper.vue Should match snapshot 1`] = `
"<div class=\\"v-stepper v-stepper--vertical theme--light\\" color=\\"deep-purple\\"><span><div class=\\"v-stepper__step v-stepper__step--active\\"><span class=\\"v-stepper__step__step deep-purple\\">1</span>
  <div class=\\"v-stepper__label\\">
    Choose a name for your event
  </div>
</div>
<transition-stub name=\\"tab-transition\\">
  <div class=\\"v-stepper__content\\">
    <div class=\\"v-stepper__wrapper\\" style=\\"height: auto;\\">
      <form novalidate=\\"novalidate\\" class=\\"v-form\\"><span><div class=\\"v-input theme--light v-text-field v-text-field--filled v-text-field--enclosed v-text-field--shaped\\"><div class=\\"v-input__control\\"><div class=\\"v-input__slot\\"><div class=\\"v-text-field__slot\\"><label for=\\"input-13\\" class=\\"v-label theme--light\\" style=\\"left: 0px; position: absolute;\\">Event name</label><input maxlength=\\"255\\" id=\\"input-13\\" type=\\"text\\"></div><div class=\\"v-input__append-inner\\"><div class=\\"v-input__icon v-input__icon--clear\\"><button disabled=\\"disabled\\" type=\\"button\\" aria-label=\\"clear icon\\" class=\\"v-icon notranslate v-icon--disabled v-icon--link mdi mdi-close theme--light\\"></button></div></div></div><div class=\\"v-text-field__details\\"><div class=\\"v-messages theme--light\\"><transition-group-stub tag=\\"div\\" name=\\"message-transition\\" class=\\"v-messages__wrapper\\"></transition-group-stub></div><div class=\\"v-counter theme--light\\">0 / 255</div></div></div></div> <button type=\\"button\\" class=\\"white--text font-weight-bold v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default deep-purple\\"><span class=\\"v-btn__content\\">
						Continue
					</span></button> <button type=\\"button\\" class=\\"v-btn v-btn--text theme--light v-size--default deep-purple--text\\"><span class=\\"v-btn__content\\">
						Cancel
					</span></button></span></form>
    </div>
  </div>
</transition-stub>
<div class=\\"v-stepper__step v-stepper__step--inactive\\"><span class=\\"v-stepper__step__step\\">2</span>
  <div class=\\"v-stepper__label\\">
    Write something about it
  </div>
</div>
<transition-stub name=\\"tab-transition\\">
  <div class=\\"v-stepper__content\\">
    <div class=\\"v-stepper__wrapper\\" style=\\"height: 0px;\\">
      <form novalidate=\\"novalidate\\" class=\\"v-form\\"><span><div class=\\"v-input v-textarea v-textarea--auto-grow v-textarea--no-resize theme--light v-text-field v-text-field--filled v-text-field--enclosed v-text-field--shaped\\"><div class=\\"v-input__control\\"><div class=\\"v-input__slot\\"><div class=\\"v-text-field__slot\\"><label for=\\"input-24\\" class=\\"v-label theme--light\\" style=\\"left: 0px; position: absolute;\\">Description</label><textarea id=\\"input-24\\" rows=\\"6\\"></textarea></div><div class=\\"v-input__append-inner\\"><div class=\\"v-input__icon v-input__icon--clear\\"><button disabled=\\"disabled\\" type=\\"button\\" aria-label=\\"clear icon\\" class=\\"v-icon notranslate v-icon--disabled v-icon--link mdi mdi-close theme--light\\"></button></div></div></div><div class=\\"v-text-field__details\\"><div class=\\"v-messages theme--light\\"><transition-group-stub tag=\\"div\\" name=\\"message-transition\\" class=\\"v-messages__wrapper\\"></transition-group-stub></div><div class=\\"v-counter theme--light\\">0</div></div></div></div> <button type=\\"button\\" class=\\"white--text font-weight-bold v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default deep-purple\\"><span class=\\"v-btn__content\\">
						Continue
					</span></button> <button type=\\"button\\" class=\\"v-btn v-btn--text theme--light v-size--default deep-purple--text\\"><span class=\\"v-btn__content\\">
						Go back
					</span></button></span></form>
    </div>
  </div>
</transition-stub>
<div class=\\"v-stepper__step v-stepper__step--inactive\\"><span class=\\"v-stepper__step__step\\">3</span>
  <div class=\\"v-stepper__label\\">
    Select an image for your event
  </div>
</div>
<transition-stub name=\\"tab-transition\\">
  <div class=\\"v-stepper__content\\">
    <div class=\\"v-stepper__wrapper\\" style=\\"height: 0px;\\">
      <div class=\\"v-image v-responsive mb-8 theme--light\\" style=\\"max-height: 350px;\\">
        <transition-stub name=\\"fade-transition\\" mode=\\"in-out\\">
          <div class=\\"v-image__image v-image__image--preload v-image__image--cover\\" style=\\"background-position: center center;\\"></div>
        </transition-stub>
        <div class=\\"v-responsive__content\\"></div>
      </div> <button type=\\"button\\" class=\\"white--text font-weight-bold v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default deep-purple\\"><span class=\\"v-btn__content\\">
				Continue
			</span></button> <button type=\\"button\\" class=\\"v-btn v-btn--text theme--light v-size--default deep-purple--text\\"><span class=\\"v-btn__content\\">
				Go back
			</span></button>
    </div>
  </div>
</transition-stub>
<div class=\\"v-stepper__step v-stepper__step--inactive\\"><span class=\\"v-stepper__step__step\\">4</span>
  <div class=\\"v-stepper__label\\">
    Tell when your event will start
  </div>
</div>
<transition-stub name=\\"tab-transition\\">
  <div class=\\"v-stepper__content\\">
    <div class=\\"v-stepper__wrapper\\" style=\\"height: 0px;\\">
      <div class=\\"row mb-8\\">
        <div class=\\"col col-6\\">
          <div class=\\"v-picker v-card v-picker--date theme--light\\">
            <div class=\\"v-picker__title deep-purple\\">
              <div class=\\"v-date-picker-title\\">
                <div class=\\"v-picker__title__btn v-date-picker-title__year\\">2021</div>
                <div class=\\"v-picker__title__btn v-date-picker-title__date v-picker__title__btn--active\\">
                  <transition-stub name=\\"picker-transition\\">
                    <div>Sat, Aug 21</div>
                  </transition-stub>
                </div>
              </div>
            </div>
            <div class=\\"v-picker__body theme--light\\" style=\\"width: 290px;\\">
              <transition-stub name=\\"fade-transition\\">
                <div>
                  <div class=\\"v-date-picker-header theme--light\\"><button type=\\"button\\" disabled=\\"disabled\\" class=\\"v-btn v-btn--disabled v-btn--icon v-btn--round theme--light v-size--default\\" aria-label=\\"Previous month\\"><span class=\\"v-btn__content\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-chevron-left theme--light\\"></i></span></button>
                    <div class=\\"v-date-picker-header__value\\">
                      <transition-stub name=\\"tab-transition\\">
                        <div class=\\"deep-purple--text\\"><button type=\\"button\\">August 2021</button></div>
                      </transition-stub>
                    </div><button type=\\"button\\" class=\\"v-btn v-btn--icon v-btn--round theme--light v-size--default\\" aria-label=\\"Next month\\"><span class=\\"v-btn__content\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-chevron-right theme--light\\"></i></span></button>
                  </div>
                  <div class=\\"v-date-picker-table v-date-picker-table--date theme--light\\">
                    <transition-stub name=\\"tab-transition\\">
                      <table>
                        <thead>
                          <tr>
                            <th>S</th>
                            <th>M</th>
                            <th>T</th>
                            <th>W</th>
                            <th>T</th>
                            <th>F</th>
                            <th>S</th>
                          </tr>
                        </thead>
                        <tbody>
                          <tr>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--text v-btn--rounded v-btn--disabled theme--light\\" disabled=\\"\\">
                                <div class=\\"v-btn__content\\">1</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--text v-btn--rounded v-btn--disabled theme--light\\" disabled=\\"\\">
                                <div class=\\"v-btn__content\\">2</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--text v-btn--rounded v-btn--disabled theme--light\\" disabled=\\"\\">
                                <div class=\\"v-btn__content\\">3</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--text v-btn--rounded v-btn--disabled theme--light\\" disabled=\\"\\">
                                <div class=\\"v-btn__content\\">4</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--text v-btn--rounded v-btn--disabled theme--light\\" disabled=\\"\\">
                                <div class=\\"v-btn__content\\">5</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--text v-btn--rounded v-btn--disabled theme--light\\" disabled=\\"\\">
                                <div class=\\"v-btn__content\\">6</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--text v-btn--rounded v-btn--disabled theme--light\\" disabled=\\"\\">
                                <div class=\\"v-btn__content\\">7</div>
                              </button></td>
                          </tr>
                          <tr>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--text v-btn--rounded v-btn--disabled theme--light\\" disabled=\\"\\">
                                <div class=\\"v-btn__content\\">8</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--text v-btn--rounded v-btn--disabled theme--light\\" disabled=\\"\\">
                                <div class=\\"v-btn__content\\">9</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--text v-btn--rounded v-btn--disabled theme--light\\" disabled=\\"\\">
                                <div class=\\"v-btn__content\\">10</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--text v-btn--rounded v-btn--disabled theme--light\\" disabled=\\"\\">
                                <div class=\\"v-btn__content\\">11</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--text v-btn--rounded v-btn--disabled theme--light\\" disabled=\\"\\">
                                <div class=\\"v-btn__content\\">12</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--text v-btn--rounded v-btn--disabled theme--light\\" disabled=\\"\\">
                                <div class=\\"v-btn__content\\">13</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--text v-btn--rounded v-btn--disabled theme--light\\" disabled=\\"\\">
                                <div class=\\"v-btn__content\\">14</div>
                              </button></td>
                          </tr>
                          <tr>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--text v-btn--rounded v-btn--disabled theme--light\\" disabled=\\"\\">
                                <div class=\\"v-btn__content\\">15</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--text v-btn--rounded v-btn--disabled theme--light\\" disabled=\\"\\">
                                <div class=\\"v-btn__content\\">16</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--text v-btn--rounded v-btn--disabled theme--light\\" disabled=\\"\\">
                                <div class=\\"v-btn__content\\">17</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--text v-btn--rounded v-btn--disabled theme--light\\" disabled=\\"\\">
                                <div class=\\"v-btn__content\\">18</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--text v-btn--rounded v-btn--disabled theme--light\\" disabled=\\"\\">
                                <div class=\\"v-btn__content\\">19</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--text v-btn--rounded v-btn--disabled theme--light\\" disabled=\\"\\">
                                <div class=\\"v-btn__content\\">20</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-date-picker-table__current v-btn--active v-btn--text v-btn--rounded theme--light deep-purple\\">
                                <div class=\\"v-btn__content\\">21</div>
                              </button></td>
                          </tr>
                          <tr>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--text v-btn--rounded theme--light\\">
                                <div class=\\"v-btn__content\\">22</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--text v-btn--rounded theme--light\\">
                                <div class=\\"v-btn__content\\">23</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--text v-btn--rounded theme--light\\">
                                <div class=\\"v-btn__content\\">24</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--text v-btn--rounded theme--light\\">
                                <div class=\\"v-btn__content\\">25</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--text v-btn--rounded theme--light\\">
                                <div class=\\"v-btn__content\\">26</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--text v-btn--rounded theme--light\\">
                                <div class=\\"v-btn__content\\">27</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--text v-btn--rounded theme--light\\">
                                <div class=\\"v-btn__content\\">28</div>
                              </button></td>
                          </tr>
                          <tr>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--text v-btn--rounded theme--light\\">
                                <div class=\\"v-btn__content\\">29</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--text v-btn--rounded theme--light\\">
                                <div class=\\"v-btn__content\\">30</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--text v-btn--rounded theme--light\\">
                                <div class=\\"v-btn__content\\">31</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--text v-btn--rounded v-btn--disabled theme--light\\" disabled=\\"\\">
                                <div class=\\"v-btn__content\\">1</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--text v-btn--rounded v-btn--disabled theme--light\\" disabled=\\"\\">
                                <div class=\\"v-btn__content\\">2</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--text v-btn--rounded v-btn--disabled theme--light\\" disabled=\\"\\">
                                <div class=\\"v-btn__content\\">3</div>
                              </button></td>
                            <td><button type=\\"button\\" class=\\"v-btn v-btn--flat v-btn--text v-btn--rounded v-btn--disabled theme--light\\" disabled=\\"\\">
                                <div class=\\"v-btn__content\\">4</div>
                              </button></td>
                          </tr>
                        </tbody>
                      </table>
                    </transition-stub>
                  </div>
                </div>
              </transition-stub>
            </div>
          </div>
        </div>
        <div class=\\"col col-6\\">
          <div class=\\"v-picker v-card v-picker--time theme--light\\">
            <div class=\\"v-picker__title deep-purple\\">
              <div class=\\"v-time-picker-title\\">
                <div class=\\"v-time-picker-title__time\\">
                  <div class=\\"v-picker__title__btn v-picker__title__btn--active\\">01</div><span>:</span>
                  <div class=\\"v-picker__title__btn\\">09</div>
                </div>
              </div>
            </div>
            <div class=\\"v-picker__body theme--light\\" style=\\"width: 290px;\\">
              <transition-stub name=\\"fade-transition\\">
                <div class=\\"v-time-picker-clock__container\\">
                  <div class=\\"v-time-picker-clock theme--light\\">
                    <div class=\\"v-time-picker-clock__inner\\">
                      <div class=\\"v-time-picker-clock__hand deep-purple\\" style=\\"transform: rotate(30deg) scaleY(1);\\"></div><span class=\\"v-time-picker-clock__item v-time-picker-clock__item--disabled\\" style=\\"left: 50%; top: 0%;\\"><span>0</span></span><span class=\\"v-time-picker-clock__item v-time-picker-clock__item--active deep-purple\\" style=\\"left: 75%; top: 6.698729810778062%;\\"><span>1</span></span><span class=\\"v-time-picker-clock__item\\" style=\\"left: 93.30127018922192%; top: 24.999999999999993%;\\"><span>2</span></span><span class=\\"v-time-picker-clock__item\\" style=\\"left: 100%; top: 50%;\\"><span>3</span></span><span class=\\"v-time-picker-clock__item\\" style=\\"left: 93.30127018922194%; top: 74.99999999999999%;\\"><span>4</span></span><span class=\\"v-time-picker-clock__item\\" style=\\"left: 75.00000000000001%; top: 93.30127018922192%;\\"><span>5</span></span><span class=\\"v-time-picker-clock__item\\" style=\\"left: 50.00000000000001%; top: 100%;\\"><span>6</span></span><span class=\\"v-time-picker-clock__item\\" style=\\"left: 25.000000000000014%; top: 93.30127018922194%;\\"><span>7</span></span><span class=\\"v-time-picker-clock__item\\" style=\\"left: 6.698729810778076%; top: 75.00000000000003%;\\"><span>8</span></span><span class=\\"v-time-picker-clock__item\\" style=\\"left: 0%; top: 50.00000000000001%;\\"><span>9</span></span><span class=\\"v-time-picker-clock__item\\" style=\\"left: 6.6987298107780475%; top: 25.000000000000032%;\\"><span>10</span></span><span class=\\"v-time-picker-clock__item\\" style=\\"left: 24.99999999999998%; top: 6.698729810778083%;\\"><span>11</span></span><span class=\\"v-time-picker-clock__item\\" style=\\"left: 49.99999999999999%; top: 19%;\\"><span>12</span></span><span class=\\"v-time-picker-clock__item\\" style=\\"left: 65.49999999999997%; top: 23.153212482682388%;\\"><span>13</span></span><span class=\\"v-time-picker-clock__item\\" style=\\"left: 76.84678751731758%; top: 34.499999999999986%;\\"><span>14</span></span><span class=\\"v-time-picker-clock__item\\" style=\\"left: 81%; top: 49.999999999999964%;\\"><span>15</span></span><span class=\\"v-time-picker-clock__item\\" style=\\"left: 76.84678751731761%; top: 65.49999999999997%;\\"><span>16</span></span><span class=\\"v-time-picker-clock__item\\" style=\\"left: 65.50000000000004%; top: 76.84678751731758%;\\"><span>17</span></span><span class=\\"v-time-picker-clock__item\\" style=\\"left: 50.000000000000014%; top: 81%;\\"><span>18</span></span><span class=\\"v-time-picker-clock__item\\" style=\\"left: 34.50000000000003%; top: 76.84678751731761%;\\"><span>19</span></span><span class=\\"v-time-picker-clock__item\\" style=\\"left: 23.153212482682427%; top: 65.50000000000004%;\\"><span>20</span></span><span class=\\"v-time-picker-clock__item\\" style=\\"left: 19%; top: 50.000000000000014%;\\"><span>21</span></span><span class=\\"v-time-picker-clock__item\\" style=\\"left: 23.153212482682385%; top: 34.50000000000003%;\\"><span>22</span></span><span class=\\"v-time-picker-clock__item\\" style=\\"left: 34.49999999999996%; top: 23.153212482682427%;\\"><span>23</span></span>
                    </div>
                  </div>
                </div>
              </transition-stub>
            </div>
          </div>
        </div>
      </div> <button type=\\"button\\" class=\\"white--text font-weight-bold v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default deep-purple\\"><span class=\\"v-btn__content\\">
				Create
			</span></button> <button type=\\"button\\" class=\\"v-btn v-btn--text theme--light v-size--default deep-purple--text\\"><span class=\\"v-btn__content\\">
				Go back
			</span></button>
    </div>
  </div>
</transition-stub></span></div>"
`;
